src/migrations/deploy-stark-proxy-v2.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  StarkProxyV2,
+  StarkProxyV2__factory,
+} from '../../types';
+import { getDeployConfig } from '../deploy-config';
+import { getDeployerSigner } from '../deploy-config/get-deployer-address';
+import { log } from '../lib/logging';
+import { waitForTx } from '../lib/util';
+
+export async function deployStarkProxyV2({
+  startStep = 0,
+  liquidityStakingAddress,
+  merkleDistributorAddress,
+  starkPerpetualAddress,
+  dydxCollateralTokenAddress,
+  numStarkProxiesToDeploy,
+
+  starkProxyNewImplAddresses,
+}: {
+  startStep?: number,
+
+  liquidityStakingAddress: string,
+  merkleDistributorAddress: string,
+  starkPerpetualAddress: string,
+  dydxCollateralTokenAddress: string,
+  numStarkProxiesToDeploy: number,
+
+  starkProxyNewImplAddresses?: string[],
+}) {
+  log('Beginning stark proxy implementation deployment\n');
+  const deployConfig = getDeployConfig();
+  const deployer = await getDeployerSigner();
+  const deployerAddress = deployer.address;
+  log(`Beginning deployment with deployer ${deployerAddress}\n`);
+
+  let starkProxyNewImpls: StarkProxyV2[] = [];
+
+  if (startStep <= 1) {
+    log('Step 1. Deploy new stark proxy implementation contracts.');
+    for (let i = 0; i < numStarkProxiesToDeploy; i++) {
+      const starkProxyNewImpl = await new StarkProxyV2__factory(deployer).deploy(
```

Comment:
I think we can deploy just a single implementation contract, and use it with all five upgradeability proxies